File: app/src/main/java/com/saizad/mvvmexample/components/main/MainActivityViewModel.java
package com.saizad.mvvmexample.components.main;

import com.saizad.mvvmexample.di.main.MainEnvironment;

import javax.inject.Inject;

public class MainActivityViewModel extends MainViewModel {

    @Inject
    public MainActivityViewModel(MainEnvironment mainEnvironment) {
        super(mainEnvironment);
    }
}

File: app/src/main/java/com/saizad/mvvmexample/components/main/MainActivityViewModel.kt
package com.saizad.mvvmexample.components.main

import androidx.lifecycle.SavedStateHandle
import com.saizad.mvvmexample.di.auth.AuthEnvironment
import com.saizad.mvvmexample.di.main.MainEnvironment
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class MainActivityViewModel @Inject constructor(
    environment: MainEnvironment
) : MainViewModel(environment)